plugins/pulp_rpm/plugins/importers/yum/repomd/modules.py

```diff
@@ -58,6 +68,24 @@ def _get_profiles(module):
     return d
 
 
+def _get_profile_defaults(module):
+    """
+    Parse stream profile defaults of a given modulemd-defaults document.
+
+    Dictionary has to be encoded due to MongoDB limitations for keys. MongoDB doesn't allow dots
+    in the keys but it's a very common case for stream names.
```

Comment:
`Profiles` often look like this:
```
{'6.0': ['default'],
 '8.0': ['super', 'another one']}
```
where key is a stream name and value is a list of default profiles.

MongoDB doesn't allow dots or dollars in dict keys.
There are few options to workaround that:
 - substitute dot with a different character (not safe, stream name can be any arbitrary string)
 - encode the whole dictionary into BSON object (current approach). The downside is that it won't be possible to retrieve or search dictionary by keys + decoding back is needed (currently it's done the serializer so search API endpoint will show decoded data)
 - encode keys only (e.g. base64) and keep the profiles field as dictionary. Possible to search by encoded key. Decoding back is needed as well.